src/math/mat4.h

```diff
@@ -8,46 +8,231 @@
 #include "math.h"
 #include <cmath>
 
+/**
+ * @class Mat4
+ * @brief 4x4 matrix class
+ */
+
 namespace Bald::Math {
     class Mat4 {
     public:
+        /**
+        * @fn               Mat4
+        * @brief            constructor
+        * @param [float]    diagonal -> diagonal value which will be used to construct matrix
+        */
         explicit constexpr Mat4(float diagonal = 1.0f);
 
-        explicit constexpr Mat4(float *data);
-
+        /**
+        * @fn               Mat4
+        * @brief            constructor
+        * @param [float*]   data -> pointer to an array of floats, which will be used to matrix
```

Comment:
@param [float*]   data -> pointer to an array of floats, which will be used to   _**construct?**_   matrix

Reply:
Thanks, corrected :+1: